pkg/genericapiserver/genericapiserver.go

```diff
@@ -385,6 +354,11 @@ func (s *GenericAPIServer) Run(options *options.ServerRunOptions) {
 	<-insecureStartedCh
 	s.RunPostStartHooks(PostStartHookContext{})
 
+	// err == systemd.SdNotifyNoSocket when not running on a systemd system
+	if err := systemd.SdNotify("READY=1\n"); err != nil && err != systemd.SdNotifyNoSocket {
```

Comment:
This was intentional. Systemd shouldn't be notified before the http port is up (once).